File: test/end-to-end/Fibonacci/C#/Generation/NonCachingGenerator.cs
using Fibonacci.Generation;

namespace Fibonacci.Generation
{

    public class NonCachingGenerator : IGenerator
    {
        public int Generate(int index)
        {
            if (index < 2)
            {
                return index;
            }

            int one = this.Generate(index - 1);
            int two = this.Generate(index - 2);

            return one + two;
        }
    }
}

File: test/end-to-end/Fibonacci/Java/Generation/NonCachingGenerator.java
package fibonacci.generation;

import fibonacci.generation.IGenerator;

public class NonCachingGenerator implements IGenerator {
    public Integer generate(Integer index) {
        if (index < 2) {
            return index;
        }

        Integer one = this.generate(index - 1);
        Integer two = this.generate(index - 2);

        return one + two;
    }
}
